rlmolecule/mcts/mcts.py

```diff
@@ -90,50 +90,50 @@ def _select(
             children = current.children
 
             if children is None:  # node is unexpanded: expand and return its value estimate
-                current.children = [self.get_vertex_for_state(state) for state in current.state.get_next_actions()]
+                self._expand(current)
```

Comment:
So in vanilla MCTS we only expand if the node has been previously visited (https://youtu.be/UXW2yZndl7U?t=107), but we're supposed to then descend to that child node in the rollout (and update its visit counts). I think this is functionally equivalent, since we expand every time, but don't count those visits for the child nodes when we `_evaluate`

Any reason you expand in `_select` rather than in `_evaluate`? I suppose this way doesn't require a subclassed evaluate to expand, but I was thinking that because AZ and MCTS expand differently, it might make sense to lump those two operations together.